### PR TITLE
Add support of auth&auth at discovery BinaryProto-lookup

### PR DESCRIPTION
**Pull request by rdhabalia**
### Motivation

Discovery BinaryProto-Lookup service

- Add support of authentication and authorization
- Avoid redirection while partitionedMetadataLookup 

### Modifications

- Authenticate client at initial connection with discovery service
- PartitionMetadataLookup and authorization 
- Remove redirection from PartitionMetadataResponse and client library

### Result

Enabling authentication and authorization at discovery lookup service

### CODE REVIEW

**Comment by yahoocla** [comment]
CLA is valid!

**Comment by yahoocla** [comment]
CLA is valid!

File: pulsar-discovery-service/src/main/java/com/yahoo/pulsar/discovery/service/server/DiscoveryServiceStarter.java [hunk 1]
```diff
@@ -78,7 +78,7 @@ public void run() {
     protected static void startWebService(ServerManager server, ServiceConfig config) throws Exception {
         // add servlet
         Map<String, String> initParameters = new TreeMap<>();
-        initParameters.put("zookeeperServers", config.getZookeeperServers());
+        initParameters.put("localZookeeperServers", config.getLocalZookeeperServers());
```
Comment (merlimat) [inline pulsar-discovery-service/src/main/java/com/yahoo/pulsar/discovery/service/server/DiscoveryServiceStarter.java]: We've being using `zookeeperServers` elsewhere for the "local ZK connection", we should not change it just here

Comment (merlimat) [inline pulsar-discovery-service/src/main/java/com/yahoo/pulsar/discovery/service/server/DiscoveryServiceStarter.java]: Also, we update the docs at https://github.com/yahoo/pulsar/blob/master/docs/ClusterSetup.md#service-discovery to add the newly introduced keys.

File: docs/BinaryProtocol.md [hunk 2]
```diff
@@ -560,14 +559,3 @@ CommandPartitionedTopicMetadataResponse {
   "partitions" : 32
 }
 ```
-
-Example of response with redirection:
```
Comment (rdhabalia) [inline docs/BinaryProtocol.md]: should we also include error response for ```TopicLookup```  and ```PartitionedMetadataLookup```

**Comment by rdhabalia** [comment]
removed ```local``` prefix from ```zookeeperserver``` config.

**Comment by rdhabalia** [comment]
@merlimat : have addressed all the changes. can you please let me know if we need any other changes.

**Pull request merged**